HackAtUCI/irvinehacks-site
# Setup admin app review pages

Closes #512 and sets up #513 

## Changes
- Updates title `IrvineHacks 2024` to `IrvineHacks 2025`
- Updates names of hacker app FRQs to properly show FRQs on hacker applicants review page
- Creates placeholder pages that shows all mentor and volunteer apps
- Adds these pages to the sidebar using [link groups](https://cloudscape.design/components/side-navigation/?tabId=playground&example=with-link-groups)
- Changes individual application page route (previously `/admin/applicants/<uid>`) to `/admin/applicants/hackers/<uid>`, `/admin/applicants/mentors/<uid>`, and `/admin/applicants/volunteers/<uid>` based on the app type
- Refactors component `Application` into 3 separate components: `HackerApplication`, `MentorApplication`, `Volunteer Application`
- Refactors component `ApplicationSection` into 3 separate components: `HackerApplicationSection`, `MentorApplicationSection`, `VolunteerApplicationSection`
- Refactors component `Applicant` to handle all 3 different app types with a prop `applicationType`
  - `Applicant` is used in `HackerApplicant`, `MentorApplicant`, and `VolunteerApplicant`
- Moves `Applicant`, `ApplicantActions`, `ApplicantOverview`, `ApplicationReviews`, and `HackerApplicantActions` to `/admin/applicants/components/` folder
- Updates `admin.py` by replacing route `/applicant/{uid}` with `/applicant/hacker/{uid}`, `/applicant/mentor/{uid}`, and `/applicant/volunteer/{uid}`
  - this decision was made so each route could be protected by user authorization roles (e.g. `Hacker Reviewer`, `Mentor Reviewer`) which will be done in a separate PR

## Testing

1. Submit multiple hacker, mentor, and volunteer apps
2. Verify that each individual application can be seen at route `/admin/applicants/<app_type>/<uid>`
